sdk/monitor/monitor-query/review/monitor-query.api.md

```diff
@@ -0,0 +1,293 @@
+    resourceId?: string;
+    // Warning: (ae-forgotten-export) The symbol "MetricUnit" needs to be exported by the entry point index.d.ts
+    unit?: MetricUnit_2;
+}
+
+// @public
+export interface MetricNamespace {
+    classification?: NamespaceClassification;
+    id?: string;
+    name?: string;
+    properties?: MetricNamespaceName;
+    type?: string;
+}
+
+// @public
+export interface MetricNamespaceName {
+    metricNamespaceName?: string;
+}
+
+// @public (undocumented)
+export class MetricsClient {
+    constructor(tokenCredential: TokenCredential, options?: MetricsClientOptions);
+    // (undocumented)
+    getMetricDefinitions(resourceUri: string, options?: GetMetricDefinitionsOptions): Promise<GetMetricDefinitionsResponse>;
+    // (undocumented)
+    getMetricNamespaces(resourceUri: string, options?: GetMetricNamespacesOptions): Promise<GetMetricNamespacesResponse>;
+    // (undocumented)
+    queryMetrics(resourceUri: string, options?: QueryMetricsOptions): Promise<QueryMetricsResponse>;
+}
+
+// @public (undocumented)
+export interface MetricsClientOptions extends PipelineOptions {
+    $host?: string;
+    endpoint?: string;
+}
+
+// @public
+export type MetricUnit = string;
+
+// @public
+export interface MetricValue {
+    average?: number;
+    count?: number;
+    maximum?: number;
+    minimum?: number;
+    timeStamp: Date;
+    total?: number;
+}
+
+// @public
+export type NamespaceClassification = string;
+
+// @public
+export type QueryGetResponse = QueryResults & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: QueryResults;
+    };
+};
+
+// @public
+export interface QueryLogsBatch {
+    queries: BatchQuery[];
+}
+
+// @public (undocumented)
+export type QueryLogsBatchOptions = OperationOptions;
+
+// @public
+export interface QueryLogsBatchResponse {
+    results?: {
+        id?: string;
+        status?: number;
+        tables?: Table[];
+        error?: ErrorInfo;
+    }[];
+}
+
+// @public (undocumented)
+export interface QueryLogsOptions extends OperationOptions {
+    includeQueryStatistics?: boolean;
+    serverTimeoutInSeconds?: number;
+    timespan?: string;
+}
+
+// @public (undocumented)
+export type QueryLogsResult = QueryResults & {
+    statistics?: QueryStatistics;
+};
+
+// @public
+export interface QueryMetricsOptions extends OperationOptions {
+    aggregations?: string[];
+    filter?: string;
+    interval?: string;
+    metricNames?: string[];
+    metricNamespace?: string;
+    orderBy?: string;
+    resultType?: ResultType;
+    timespan?: string;
+    top?: number;
+}
+
+// @public
+export interface QueryMetricsResponse {
+    cost?: number;
+    interval?: string;
+    metrics: Metric[];
+    namespace?: string;
+    resourceRegion?: string;
+    timespan: string;
+}
+
+// @public
+export interface QueryResults {
+    statistics?: any;
+    tables: Table[];
+}
+
+// @public (undocumented)
+export interface QueryStatistics {
```

Comment:
is it worth making this a generic? Or is it really open ended in a way that's unknowable?

Reply:
I think we're going to do some more discussion about this. Most of the other teams are just leaving it completely untyped, but I'd like to see if we can actually go the other direction and add some more fields. 

There might still be some unmapped, however, but we can cross that bridge in between this release and the next, I think.

Reply:
btw, I have this covered in my "pre preview" issue: https://github.com/Azure/azure-sdk-for-js/issues/15296